Tests/TestingTests/PollingTests.swift

```diff
@@ -0,0 +1,352 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@testable @_spi(Experimental) @_spi(ForToolsIntegrationOnly) import Testing
+
+@Suite("Polling Tests")
+struct PollingTests {
+  @Suite("PollingBehavior.passesOnce")
+  struct PassesOnceBehavior {
+    let delta = Duration.seconds(6)
+
+    @Test("Simple passing expressions") func trivialHappyPath() async {
+      await #expect(until: .passesOnce) { true }
+
+      await #expect(until: .passesOnce, throws: PollingTestSampleError.ohNo) {
+        throw PollingTestSampleError.ohNo
+      }
+
+      await #expect(until: .passesOnce, performing: {
+        throw PollingTestSampleError.secondCase
+      }, throws: { error in
+        (error as? PollingTestSampleError) == .secondCase
+      })
+
+      await #expect(until: .passesOnce, throws: PollingTestSampleError.ohNo) {
+        throw PollingTestSampleError.ohNo
+      }
+    }
+
+    @Test("Simple failing expressions") func trivialSadPath() async {
+      await confirmation("Polling failed", expectedCount: 1) { failed in
+        var configuration = Configuration()
+        configuration.eventHandler = { event, _ in
+          if case .issueRecorded = event.kind {
+            failed()
+          }
+        }
+        await Test {
+          await #expect(until: .passesOnce) { false }
+        }.run(configuration: configuration)
+      }
```

Comment:
I'm going to write a test helper to make it much easier to verify that issues were reported, as `withKnownIssue` is the wrong semantics for what I want. I'm also tempted to make it available under `ForToolsIntegrationOnly`, but I also think that should be a separate pitch?

I really want something that I can use like:

```swift
let issues = await collectIssues {
    #expect(Bool(false))
}
```